File: 599-minimum-index-sum-of-two-lists/599-minimum-index-sum-of-two-lists.cpp
class Solution {
public:
    vector<string> findRestaurant(vector<string>& list1, vector<string>& list2) {
        unordered_map<string,int> hmap;
        int minSum{INT_MAX},sum;
        vector<string> res;
        for(int i=0;i<list1.size();i++){
           hmap.insert(make_pair(list1[i],i));
        }
        for(int i=0;i<list2.size();i++){
            if(hmap.find(list2[i])!=hmap.end()){
                sum=i+hmap.find(list2[i])->second;
                if(sum<minSum){
                    res.clear();
                    res.push_back(list2[i]);
                    minSum=sum;
                }
                else if(sum==minSum){
                     res.push_back(list2[i]);
                }
            }
        }
        return res;
    }
};

File: 599-minimum-index-sum-of-two-lists/599-minimum-index-sum-of-two-lists.py
class Solution:
    def findRestaurant(self, list1: List[str], list2: List[str]) -> List[str]:
        hmap, ans, minSum={},[],float('inf')
        for i,res in enumerate(list1):
            hmap[res]=i
        for i,res in enumerate(list2):
            if(res in hmap):
                lSum=hmap[res]+i
                if(lSum<minSum):
                    ans.clear()
                    ans.append(res)
                    minSum=lSum
                elif(lSum==minSum):
                    ans.append(res)
        return ans